crates/wit-component/src/encoding.rs

```diff
@@ -1243,6 +1422,33 @@ impl<'a> EncodingState<'a> {
         Ok(())
     }
 
+    fn payload_type_index(&mut self, ty: TypeId, imported: bool) -> Result<u32> {
```

Comment:
Mind adding a comment here explaining what this function is doing? I think a comment on `owner` might also be reasonable to explain why it's traversing the way it is